actors/organizer.go

```diff
@@ -0,0 +1,407 @@
+package actors
+
+import (
+	"encoding/json"
+	"student20_pop/db"
+	"student20_pop/define"
+	"fmt"
+)
+
+type Organizer struct {
+	PublicKey string
+	database  string
+}
+
+func NewOrganizer(pkey string, db string) *Organizer {
+	return &Organizer{
+		PublicKey: pkey,
+		database:  db,
+	}
+}
+
+// Test json input to create LAO:
+//  careful with base64 needed to remove
+//  careful with comma after witnesses[] and witnesses_signatures[] needed to remove
+
+/** processes what is received from the WebSocket
+ * msg : receivedMessage
+ * returns, in order :
+ * message to send on channel, or nil
+ * channel for the message, or nil
+ * response to the sender, or nil
+ */
+func (o *Organizer) HandleWholeMessage(msg []byte, userId int) ([]byte, []byte, []byte) {
+	generic, err := define.AnalyseGeneric(msg)
+	if err != nil {
+		fmt.Printf("1")
+		return nil, nil, define.CreateResponse(define.ErrRequestDataInvalid, nil, generic)
+	}
+
+	var history []byte = nil
+	var message []byte = nil
+	var channel []byte = nil
+
+	switch generic.Method {
+	case "subscribe":
+		message, channel, err = nil, nil, handleSubscribe(generic, userId)
+	case "unsubscribe":
+		message, channel, err = nil, nil, handleUnsubscribe(generic, userId)
+	case "publish":
+		message, channel, err = o.handlePublish(generic)
+	//case "message": err = h.handleMessage()
+	//Potentially, we never receive a "message" and only output "message" after a "publish" in order to broadcast.
+	//Or they are only notification, and we just want to check that it was a success
+	case "catchup":
+		history, err = o.handleCatchup(generic)
+	default:
+		fmt.Printf("2")
```

Comment:
There are lots of theses around in the code, we changed this yet the current version